### PR TITLE
573 allow to add remove elements from catalog dynamically through api

### PR DESCRIPTION
**Pull request by apupier**

### CODE REVIEW

File: camel-kafka-connector-catalog/src/main/java/org/apache/camel/kafkaconnector/catalog/CamelKafkaConnectorCatalog.java [hunk 1]
```diff
@@ -83,7 +83,7 @@ private void initCatalog() {
         }
     }
 
-    public String getConnectorAsJson(String connectorName) {
+    private String loadConnectorAsJsonFromEmbeddedCatalog(String connectorName) {
```
Comment (oscerd) [inline camel-kafka-connector-catalog/src/main/java/org/apache/camel/kafkaconnector/catalog/CamelKafkaConnectorCatalog.java:86]: I wouldn't change this name. It doesn't add so much.

File: camel-kafka-connector-catalog/src/test/java/org/apache/camel/kafkaconnector/catalog/CamelKafkaConnectorCatalogTest.java [hunk 2]
```diff
@@ -20,30 +20,30 @@
 import java.util.Map;
 
 import org.apache.camel.kafkaconnector.model.CamelKafkaConnectorModel;
+import org.apache.camel.kafkaconnector.model.CamelKafkaConnectorOptionModel;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (oscerd) [inline camel-kafka-connector-catalog/src/test/java/org/apache/camel/kafkaconnector/catalog/CamelKafkaConnectorCatalogTest.java:27]: Please don't import with * operator

**Review by oscerd — COMMENTED** [review]
Just two things: the method name is not very important. The import could be easier and just use the needed classes.

File: camel-kafka-connector-catalog/src/main/java/org/apache/camel/kafkaconnector/catalog/CamelKafkaConnectorCatalog.java (hunk 1 above)
Comment (apupier) [reply camel-kafka-connector-catalog/src/main/java/org/apache/camel/kafkaconnector/catalog/CamelKafkaConnectorCatalog.java:86]: The added difference is that the method is loading from embedded catalog, it is not able to provide the json from the cached connector that are dynamically added. With the name "getConnectorAsJson", I'm expecting to have also the dynamically added connector to be provided.

File: camel-kafka-connector-catalog/src/test/java/org/apache/camel/kafkaconnector/catalog/CamelKafkaConnectorCatalogTest.java (hunk 2 above)
Comment (apupier) [reply camel-kafka-connector-catalog/src/test/java/org/apache/camel/kafkaconnector/catalog/CamelKafkaConnectorCatalogTest.java:27]: I was only using the defaults of Eclipse projects but fine to follow another rule.

File: camel-kafka-connector-catalog/src/main/java/org/apache/camel/kafkaconnector/catalog/CamelKafkaConnectorCatalog.java (hunk 1 above)
Comment (oscerd) [reply camel-kafka-connector-catalog/src/main/java/org/apache/camel/kafkaconnector/catalog/CamelKafkaConnectorCatalog.java:86]: Sounds good them

**Comment by oscerd** [comment]
LGTM. Let me merge.

**Pull request merged**